programs/cleos/CLI11.hpp

```diff
@@ -2434,3 +2434,40 @@ struct AppFriend {
 } // namespace detail
 
 } // namespace CLI
+
+namespace eosio {
+   namespace cleos {
+   struct KeosdSignalHndl {
+      enum state : size_t
+      {
+         KEOSD_NOT_READY = 0,
+         KEOSD_READY = 1
+      };
+
+      static void init() {
+         ready_signal_received = KEOSD_NOT_READY;
+         std::signal(SIGUSR1, handle_keosd_start_signal);
+      }
+
+      static void handle_keosd_start_signal(int) {
+         ready_signal_received = KEOSD_READY;
```

Comment:
This causes a data race.  See [intro.races]
"Two actions are potentially concurrent if... they are unsequenced, at least one is performed by a signal handler, and they are not both performed by the same signal handler invocation."

"The execution of a program contains a data race if it contains two potentially concurrent conflicting actions, at least one of which is not atomic, and neither happens before the other..."

Reply:
Cool, thanks, it is good reading.